packages/cli/src/handlers/oauthLogin.ts

```diff
@@ -0,0 +1,374 @@
+import {
+    AuthorizationError,
+    CreatePersonalAccessToken,
+    formatDate,
+    oauthPageStyles,
+} from '@lightdash/common';
+import { exec } from 'child_process';
+import * as crypto from 'crypto';
+import * as http from 'http';
+import { createServer } from 'net';
+import fetch from 'node-fetch';
+import { URL } from 'url';
+import { promisify } from 'util';
+import GlobalState from '../globalState';
+import * as styles from '../styles';
+
+// Shared CSS styles for OAuth response pages - matching Lightdash login design
+
+const execAsync = promisify(exec);
+
+// PKCE helper functions
+const generateCodeVerifier = (): string =>
```

Comment:
we should use a library for all of this, we typically use the openid-client library, which contains all of this logic

Reply:
Done in another PR